File: src/VBNetFx/FacInterfaces/Legatro/LegatroTimeDataImport.vb
Imports System.Windows.Forms

<FacessoImportFilterName("Legato-Zeitdatenimport", FacessoImportType.TimeKeepingData, FacessoInterfaceBrand.LegatroTimeKeeping)> _
Public Class LegatroTimeDataImport
    Inherits TimeDataImportBase

    Private myLegatroSQLConnectionString As String

    Public Overrides Function ConfigureGenericInterface() As System.Windows.Forms.DialogResult
        Return MessageBox.Show("Configure Inport Filter")
    End Function

    Public Overrides Function ConfigureImportFilter() As System.Windows.Forms.DialogResult
        Dim frm As New frmLegatroTimeDataConfigDialog
        Return frm.HandleDialog(Me)
    End Function

    Public Overrides Function GetData(ByVal ProductionDate As Date, ByVal Shift As ShiftCombination) As IImportResultTable
        Dim ltdt As New LegatroTimeDataTransformation(ProductionDate, Shift, Me)
        ltdt.Convert()
        Return ltdt.ResultTable
    End Function

    Public Overrides ReadOnly Property InterfaceBrand() As FacessoInterfaceBrand
        Get
            Return FacessoInterfaceBrand.LegatroTimeKeeping
        End Get
    End Property

    Public Property LegatroSQLConnectionString() As String
        Get
            Return myLegatroSQLConnectionString
        End Get
        Set(ByVal value As String)
            myLegatroSQLConnectionString = value
        End Set
    End Property

    Public Overrides Function AssembleConversionItems() As FacessoConversionItemsBase
        If String.IsNullOrEmpty(LegatroSQLConnectionString) Then
            Return Nothing
        End If

        Dim locConversionItems As New FacessoConversionItemsBase

        Dim dc As New LegatroDataContext(LegatroSQLConnectionString)
        Dim wg = From wgItems In dc.WorksitesOrProjects _
               Order By wgItems.WorkEntityNumber
        For Each wgItem In wg
            locConversionItems.Add(New FacessoConversionItemBase(wgItem.WorkEntityNumber, wgItem.WorkEntityName))
        Next
        Return locConversionItems
    End Function

    Public Overrides ReadOnly Property IsGenericInterfaceConfigured() As Boolean
        Get
            Return True
        End Get
    End Property


End Class

File: src/NetFxCSharp/Facesso.Interfaces/Legatro/LegatroTimeDataImport.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso.Interfaces
{
    [FacessoImportFilterName("Legato-Zeitdatenimport", FacessoImportType.TimeKeepingData, FacessoInterfaceBrand.LegatroTimeKeeping)]
    public class LegatroTimeDataImport : TimeDataImportBase
    {
        private string myLegatroSQLConnectionString;
        public override System.Windows.Forms.DialogResult ConfigureGenericInterface()
        {
            return MessageBox.Show("Configure Inport Filter");
        }

        public override System.Windows.Forms.DialogResult ConfigureImportFilter()
        {
            frmLegatroTimeDataConfigDialog frm = new frmLegatroTimeDataConfigDialog();
            return frm.HandleDialog(this);
        }

        public override IImportResultTable GetData(System.DateTime ProductionDate, ShiftCombination Shift)
        {
            LegatroTimeDataTransformation ltdt = new LegatroTimeDataTransformation(ProductionDate, (int)(Shift), this);
            ltdt.Convert();
            return ltdt.ResultTable();
        }

        public override FacessoInterfaceBrand InterfaceBrand
        {
            get
            {
                return FacessoInterfaceBrand.LegatroTimeKeeping;
            }
        }

        public string LegatroSQLConnectionString
        {
            get
            {
                return myLegatroSQLConnectionString;
            }

            set
            {
                myLegatroSQLConnectionString = value;
            }
        }

        public override FacessoConversionItemsBase AssembleConversionItems()
        {
            if (string.IsNullOrEmpty(LegatroSQLConnectionString))
            {
                return null;
            }

            FacessoConversionItemsBase locConversionItems = new FacessoConversionItemsBase();
            LegatroDataContext dc = new LegatroDataContext(LegatroSQLConnectionString);
            var wg = (
                from wgItems in dc.WorksitesOrProjects
                orderby wgItems.WorkEntityNumber
                select wgItems);
            foreach (var wgItem in wg)
            {
                locConversionItems.Add(new FacessoConversionItemBase(wgItem.WorkEntityNumber, wgItem.WorkEntityName));
            }

            return locConversionItems;
        }

        public override bool IsGenericInterfaceConfigured
        {
            get
            {
                return true;
            }
        }
    }
}